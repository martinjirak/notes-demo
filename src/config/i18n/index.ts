import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import xhr from 'i18next-xhr-backend'
import { initReactI18next } from 'react-i18next'

const options: i18n.InitOptions = {
    load: 'languageOnly',
    ns: ['global'],
    fallbackLng: 'en',
    defaultNS: 'global',
    backend: {
        loadPath: 'public/locales/{{lng}}/{{ns}}.json',
    },
    detection: {
        order: ['localStorage', 'navigator'],
        caches: ['localStorage'],
        lookupLocalStorage: 'i18nextLng',
    },
    interpolation: {
        escapeValue: false,
    },
    react: {
        bindI18n: 'languageChanged',
    },
}

export default i18n.use(LanguageDetector).use(xhr).use(initReactI18next).init(options)
