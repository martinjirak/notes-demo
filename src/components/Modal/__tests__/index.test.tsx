import * as React from 'react'
import { createMount } from '@material-ui/core/test-utils'
import { I18nextProvider } from 'react-i18next'
import i18n from '../../../config/tests/i18n/index'
import { Modal, ModalProps } from '../'

describe('Modal should match Snapshot', () => {
    it('without hint', () => {
        const props: ModalProps = {
            onClose: jest.fn(),
            children: <>child</>,
        }

        const renderedValue = createMount()(
            <I18nextProvider i18n={i18n}>
                <Modal {...props} />
            </I18nextProvider>
        )
        expect(renderedValue.html()).toMatchSnapshot()
    })
})
