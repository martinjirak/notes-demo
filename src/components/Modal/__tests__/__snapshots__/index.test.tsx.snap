// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match Snapshot without hint 1`] = `"<div role=\\"presentation\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div aria-hidden=\\"true\\" style=\\"z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"makeStyles-paper-1\\" tabindex=\\"-1\\">child</div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;
